.clang-tidy

```diff
@@ -0,0 +1,112 @@
+---
+Checks: >
```

Comment:
Many of these checks are really useful, but I disabled those which were non trivial to resolve at the moment, but I propose to resolve them afterwards iteratively in separate PRs.

Reply:
Do you propose to fix them as part of this PR? Or it is just an advice for future PRs?

Reply:
Most of the straight-forward fixes can be merged in a separate PR. Then this one can be rebased.